### PR TITLE
Webhooks

### PR DESCRIPTION
**Pull request by jhadobe**
## Purpose of this pull request

This pull request (PR) adds a webhook configuration example to the starter kit docs.

### CODE REVIEW

File: src/pages/app-development/starter-kit/hooks.md [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+ - App Builder
+ - API Mesh
+ - Events
+ - REST
+ - Tools
+---
+
+# Webhooks example
+
+The following runtime action demonstrates how to expose an entry point to perform a realtime check of the available stock for Commerce cart items. You must implement the stock-checking logic on an external service.
```
Comment (rojoangel) [inline src/pages/app-development/starter-kit/hooks.md:15]: I find the statement `The following runtime action` a bit misleading as there is no `runtime action` after this statement. Maybe it could be rephrased to something like:
> The runtime action at `actions/webhook/check-stock/index.js` demonstrates ...

File: src/pages/app-development/starter-kit/hooks.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  statusCode: '200',
+  body: {
+    op: 'success'
+  }
+}
+```
+
+<InlineAlert variant="info" slots="text"/>
+
+These responses are adapted to the [Commerce webhook module](https://developer.adobe.com/commerce/extensibility/webhooks/). You can modify the response implementation in the code as needed.
```
Comment (rojoangel) [inline src/pages/app-development/starter-kit/hooks.md:119]: Can we link directly to the "Webhooks responses & logging" responses section instead?
```suggestion
These responses are adapted to the [Commerce webhook module](https://developer.adobe.com/commerce/extensibility/webhooks/responses/#responses). You can modify the response implementation in the code as needed.
```

File: src/pages/app-development/starter-kit/hooks.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+ - Events
+ - REST
+ - Tools
+---
+
+# Webhooks example
+
+The following runtime action demonstrates how to expose an entry point to perform a realtime check of the available stock for Commerce cart items. You must implement the stock-checking logic on an external service.
+
+This runtime response uses the [Adobe Commerce webhook module](https://developer.adobe.com/commerce/extensibility/webhooks/) to allow intercepting flows in Adobe Commerce.
```
Comment (rojoangel) [inline src/pages/app-development/starter-kit/hooks.md:17]: This document already covers responses at the end, maybe this statement could be rephrased to
>  This runtime **action uses the approach described at** [Adobe Commerce webhook module](https://developer.adobe.com/commerce/extensibility/webhooks/) to allow intercepting flows in Adobe Commerce.

File: src/pages/app-development/starter-kit/hooks.md [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    final: true
+```
+
+## Responses
+
+Error response:
+
+```javascript
+return {   
+  statusCode: '200',
```
Comment (rojoangel) [inline src/pages/app-development/starter-kit/hooks.md:98]: Typo: the quotes are not needed
```suggestion
  statusCode: 200,
```

File: src/pages/app-development/starter-kit/hooks.md [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+    message: 'Error message'
+  }
+}
+```
+
+Success response:
+
+```javascript
+return {
+  statusCode: '200',
```
Comment (rojoangel) [inline src/pages/app-development/starter-kit/hooks.md:110]: Typo: the quotes are not needed
```suggestion
  statusCode: 200,
```

**Review by rojoangel — APPROVED** [review]
I suggest adding a statement at the beginning of the doc to explain that the starter kit supports webhooks and that this is an example/reference implementation about how to use them to check stock availability.

Good job!

**Comment by jhadobe** [comment]
@rojoangel can you re-review? I've rearranged the webhooks example and added ingestion webhooks coverage.
Additionally, can you tell me what set up in the Adobe Commerce admin is required to get hooks to work? (outside of initial setup of the Starter Kit).

File: src/pages/app-development/starter-kit/webhooks-example.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+- `data.cart_id`
+- `data.items`
+
+Sample validation logic is provided in the following location in the Starter Kit package.
+
+- `webhook\check-stock\validator.js`
+
+You can modify this file to add more validations to the `validateData` method, if necessary.
+
+### Authentication
```
Comment (rojoangel) [inline src/pages/app-development/starter-kit/webhooks-example.md:66]: The title of the section seems to be unrelated to its content. It could maybe be changed to `Business logic` or something similar.

**Review by rojoangel — APPROVED** [review]
Looks good to me.

File: src/pages/app-development/starter-kit/webhook-ingestion.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+keywords:
+ - Extensibility
+ - App Builder
+ - API Mesh
+ - Events
+ - REST
+ - Tools
+---
+
+# Webhooks ingestion
```
Comment (rojoangel) [inline src/pages/app-development/starter-kit/webhook-ingestion.md:13]: Maybe this has to be the other way around: `Ingestion webhook`: it is a webhook that allows the ingestion of events

File: src/pages/app-development/starter-kit/webhook-ingestion.md [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+```yaml
+ingestion:
+  license: Apache-2.0
+  actions:
+    $include: ./actions/ingestion/actions.config.yaml
+```
+
+## Customizing ingest webhooks
+
+Data parameters contain the array of [events](../../events/) to publish. Each event must include an entity, `event`, and `value`. The value parameter contains the data to send through the event.
```
[the remark this answers is not in the document]
Comment (jhadobe) [reply src/pages/app-development/starter-kit/webhook-ingestion.md:30]: @hajji

**Review by ahajji87 — APPROVED** [review]

File: src/pages/app-development/starter-kit/webhooks-example.md [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+
+<InlineAlert variant="info" slots="text"/>
+
+You must implement the stock-checking logic on an external service.
+
+## Configure webhook
+
+This runtime action uses the [Adobe Commerce webhook module](../../webhooks/) to allow intercepting flows in Adobe Commerce.
+
+Follow [configure hooks](https://developer.adobe.com/commerce/extensibility/webhooks/hooks/) to modify your `webhooks.xml` file and define the connection between Adobe Commerce and your backoffice system using the `observer.checkout_cart_product_add_before` method.
```
Comment (keharper) [inline src/pages/app-development/starter-kit/webhooks-example.md:33]: Someone in Starter Kit won't have a reference webhooks file. They'll have to manually configure the webhook in the Admin. Change the link target to Admin configuration, and I'll need to update the top of that article to be more inclusive for brand new developers.

**Review by keharper — APPROVED** [review]
